common/djangoapps/student/views.py

```diff
@@ -426,9 +426,9 @@ def _cert_info(user, course_overview, cert_status, course_mode):  # pylint: disa
         persisted_grade = CourseGradeFactory().read(user, course=course_overview)
         if persisted_grade is not None:
             status_dict['grade'] = unicode(persisted_grade.percent)
-        elif 'grade' in cert_status:
-            status_dict['grade'] = cert_status['grade']
-        else:
+        if 'grade' not in status_dict or int(cert_status.get('grade', '0')) > persisted_grade.percent:
```

Comment:
If `persisted_grade` is `None`, then line 428 will not have been executed and `'grade' not in status_dict` will return `True` and short-circuit the `if` statement. That said, your cleaned-up version is a lot better. You should have seen the original version of this I had before the first round of cleanup!